public_html/wp-content/plugins/wordcamp-speaker-feedback/assets/js/script.js

```diff
@@ -4,13 +4,60 @@
 ( function() {
 	function onFormNavigate( event ) {
 		event.preventDefault();
-		const value = event.target[ 0 ].value;
+		var value = event.target[ 0 ].value;
 		// Use the fact that post IDs will redirect to the right page.
 		window.location = '/?p=' + value + '&sft_feedback=1';
 	}
 
-	const form = document.getElementById( 'sft-navigation' );
-	if ( form ) {
-		form.addEventListener( 'submit', onFormNavigate, true );
+	function onFormSubmit( event ) {
+		event.preventDefault();
+		var rawData = new FormData( event.target );
+		var data = {
+			post: rawData.get( 'sft-post' ),
+			meta: {
+				rating: rawData.get( 'sft-rating' ),
+				q1: rawData.get( 'sft-question-1' ),
+				q2: rawData.get( 'sft-question-2' ),
+				q3: rawData.get( 'sft-question-3' ),
+			},
+		};
+
+		var author = rawData.get( 'sft-author' );
+		if ( '0' !== author ) {
+			data.author = author;
+		} else {
+			data.author_name = rawData.get( 'sft-author-name' );
+			data.author_email = rawData.get( 'sft-author-email' );
+		}
+
+		var messageContainer = document.getElementById( 'speaker-feedback-notice' );
+		// Reset the notice before submission.
+		messageContainer.setAttribute( 'class', '' );
+		messageContainer.innerText = '';
+
+		wp.apiFetch( {
+			path: '/wordcamp-speaker-feedback/v1/feedback',
+			method: 'POST',
+			data: data,
+		} )
+			.then( function() {
+				messageContainer.setAttribute( 'class', 'speaker-feedback__notice is-success' );
+				messageContainer.innerText = 'Feedback submitted.';
+				event.target.replaceWith( messageContainer );
+			} )
+			.catch( function( error ) {
+				messageContainer.setAttribute( 'class', 'speaker-feedback__notice is-error' );
+				messageContainer.innerText = error.message;
+			} );
```

Comment:
The API response when there's a meta validation error includes a `data` attribute that contains the key of each field that was invalid and an explanation. Would it make sense to output those as well?

Example, when you skip the rating:

Instead of just "One or more fields has invalid data." it could be

```
One or more fields has invalid data.
* rating: This value is too low
```

Reply:
Added these error messages to their relevant inputs in a1b0be0c54ae2ab57bf493bbd2b31a22d77a7532 & updated the screenshot in the description.